### DISCUSSION TITLE
How to debug libpod

### DISCUSSION THREAD

**Discussion by BlackHole1** [containers/podman#18935] [Q&A] [answered]
I can print detailed logs of the podman client by using `podman --debug`, but I cannot see the logs of the podman server (libpod). I want to know if there is any way to view the logs in libpod.

**Comment by vrothberg** [comment, marked as answer]
Thanks for reaching out, @BlackHole1.

By default, the server is running via the socket-activated `podman.service` systemd service and runs with `--log-level=info`.  If you want to see debug logs you could do a `sudo systemctl edit podman.service` and edit specific line setting the log level.  Alternatively, you can spin up a server on your own via `podman --log-level=debug system service -t0`.

**Reply by BlackHole1** [reply]
Thank you for your reply! @vrothberg 
When I run `podman --log-level=debug system service -t0`, I can confirm that this is exactly what I need!

**Comment by BlackHole1** [comment]
Hey @vrothberg 👋. Sorry to bother you again 🙇‍♂️. I am currently making modifications to the code related to `libpod`, but the changes are not taking effect. This is because podman running in `Fedora CoreOS` is built-in and not locally built by me. I would like to understand how the podman team develops/debugs libpod locally.

**Reply by BlackHole1** [reply]
My initial idea was to build a local Linux platform that is not a remote podman. Then, I would disable the podman service in `Fedora CoreOS` and replace it with the locally built podman. However, I encountered difficulties when trying to execute `make podman`.

<img width="2645" alt="image" src="https://github.com/containers/podman/assets/8198408/d68f65b8-2265-418a-afe5-a3cc27c66ed8">

**Reply by vrothberg** [reply]
The screenshot indicates that you're building on a Mac.  There, you need to `make podman-remote`.

**Reply by BlackHole1** [reply]
Is Mac currently only supporting podman-remote? If I want to build podman, can I only do it on Ubuntu?

**Reply by BlackHole1** [reply]
It seems that Podman can only be built on Linux. I have successfully built it on Ubuntu.

<img width="783" alt="image" src="https://github.com/containers/podman/assets/8198408/6a2a6839-938e-41bd-beba-5b4cc32a3fc2">

**Reply by vrothberg** [reply]
Yes, you need all the libraries to build it.  It's not a static binary.

**Comment by rhatdan** [comment]
The full podman executable can only be built on Linux and FreeBSD.  Podman-remote is what we build for non UNIX platforms.

**Reply by BlackHole1** [reply]
Thank you for your reply!
